lib/node_modules/@stdlib/lapack/base/dlaset/lib/base.js

```diff
@@ -80,17 +81,30 @@ function setDiagonal( M, N, beta, A, strideA1, strideA2, offsetA ) {
 * var A = new Float64Array( [ 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0 ] );
 *
 * setUpper( 2, 3, 3.1, 3.7, A, 3, 1, 0 );
-* // A => <Float64Array>[ 3.7, 0.0, 3.1, 0.0, 3.7, 0.0 ]
+* // A => <Float64Array>[ 3.7, 3.1, 0.0, 0.0, 3.7, 0.0 ]
 */
 function setUpper( M, N, alpha, beta, A, strideA1, strideA2, offsetA ) {
 	var ia;
 	var i;
 	var j;
 
+	if ( isRowMajor( [ strideA1, strideA2 ] ) ) {
+		for ( i = 0; i < M; i++ ) {
+			ia = offsetA + ( i * strideA1 );
+			for ( j = i + 1; j < min( M, N ); j++ ) {
+				if ( i !== j ) {
```

Comment:
@Pranavchiku Given that `j = i + 1`, in what circumstance is `i` equal to `j`?

Reply:
Interesting, it fails without that check.